src/services/course.service.ts

```diff
@@ -0,0 +1,25 @@
+import { createError } from "../hooks";
+import { CONFLICT } from "../types/errors.type";
+import { Course } from "../models";
+import { addCourse_I } from "../types/course.type";
+
+export class CourseService {
+    static async addCourse(requestBody: addCourse_I) {
+        const isCourseExist = await Course.findOne({ title: requestBody.title });
+        if (isCourseExist) {
+            throw createError(
+                'COURSE_ALREADY_EXISTS',
+                400,
+                'Course already exists'
+            );
+        }
```

Comment:
also validate instructor exists or not in users 

Please write BAD_REQUEST insted of COURSE_ALREADY_EXISTS